### ISSUE TITLE
Group Policy for Storage doesn't allow to list keys

### ISSUE THREAD

**Issue by DanielMuller** [aws-amplify/amplify-cli#2700]
**Describe the bug**
After adding a new group to Cognito and updating the storage to allow members of this group to read, write and delete, the generate IAM policy doesn't allow to list objects.

The reason is that the generated policy is:
```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "s3:PutObject",
        "s3:GetObject",
        "s3:ListBucket",
        "s3:DeleteObject"
      ],
      "Resource": [
        "arn:aws:s3:::myStorageBucket/*"
      ],
      "Effect": "Allow"
    }
  ]
}
```

`ListBucket` needs a privilege on a bucket, not a key. When manually changing the policy (via web console) to:
```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "s3:PutObject",
        "s3:GetObject",
        "s3:DeleteObject"
      ],
      "Resource": [
        "arn:aws:s3:::myStorageBucket/*"
      ],
      "Effect": "Allow"
    },
    {
      "Action": [
        "s3:ListBucket"
      ],
      "Resource": [
        "arn:aws:s3:::myStorageBucket"
      ],
      "Effect": "Allow"
    }
  ]
}
```
Listing objects works.

I identified the source code generating this here: https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-category-storage/provider-utils/awscloudformation/cloudformation-templates/s3-cloudformation-template.json.ejs#L269-L291
But I don't think I am able to provide a PR, since I don't really understand how all this works.

**Amplify CLI Version**
3.17.0

**To Reproduce**
`amplify auth update` => Create a new Group
`amplify storage update` => Grant Read access to the group on the S3 bucket

**Expected behavior**
Allow `s3:ListBucket` on `"arn:aws:s3:::myStorageBucket"`

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Node Version: 10.16.3

**Additional context**
Object listing is done via aws-sdk in a Browser, using credentials from `Auth.essentialCredentials` with `s3.listObjectsV2`

**Comment by BardiaN** [comment]
+1

Spent quite some time to figured it out today. :P

**Comment by kaustavghosh06** [comment]
We've merged the Pr - #3612 with a fix for this. It'll be part of our next release.
